### PR TITLE
Histogram function implemented and tests added

### PR DESCRIPTION
**Pull request by khalidcawl**

### CODE REVIEW

File: src/slimeda/histogram.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
+import pandas as pd
 
 def histogram(df, columns, path=""):
     """
-    Creates histogram chart objects for specific columns in a data frame
+    Creates histogram chart objects for specific columns in a data frame and optionally saves
+    the charts to a user-specific path
     Parameters
     ----------
     df: pd.DataFrame
```
Comment (Anthea98) [inline src/slimeda/histogram.py:10]: do we need an explanation for the path?

File: tests/test_histogram.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    # create a dummy data frame with four columns for testing
+    df = pd.DataFrame(rng.integers(0, 100, size=(100, 4)), columns=list('ABCD'))
+    charts = histogram(df, ["A"])
+    assert charts != None
+    assert len(charts) == 1
+    assert charts[0] != None
+    assert charts[0].data.equals(df)
+    assert charts[0].mark == "bar"
+
```
Comment (Anthea98) [inline tests/test_histogram.py:17]: According to the requirements: "Write a unit test function for each specified function, which is named after the function being tested (e.g., if the function is named foo then the unit test function is named test_foo).  The unit test function should test 3 to 5 edge cases to ensure that the function returns what is expected to the user. " maybe rename the first test function and also I am not quite clear what should be the edge cases in this test (as for me, I think the tests below are representative enough)

File: tests/test_histogram.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    with pytest.raises(ValueError):
+        rng = np.random.default_rng()
+        df = pd.DataFrame(rng.integers(0, 100, size=(100, 4)), columns=list('ABCD'))
+        histogram(df, [])
+
+def test_histogram_wrong_column():
+    with pytest.raises(ValueError):
+        rng = np.random.default_rng()
+        df = pd.DataFrame(rng.integers(0, 100, size=(100, 4)), columns=list('ABCD'))
+        histogram(df, ["E"])
```
Comment (Anthea98) [inline tests/test_histogram.py:45]: Nice work! But I think we need some test docs to briefly explain for each test~

**Review by Anthea98 — CHANGES_REQUESTED** [review]
Nice work! Here are some suggestions for details~

**Comment by thayeylolu** [comment]
@khalidcawl  How are you ci/cd testing not failing ?

File: src/slimeda/histogram.py (hunk 1 above)
Comment (khalidcawl) [reply src/slimeda/histogram.py]: Hi. Thanks for catching that. I was planning on removing that parameter. I think we can leave out saving the images to local folders. This would be a bit tricky to test. We don't have to have that.

File: tests/test_histogram.py (hunk 2 above)
Comment (khalidcawl) [reply tests/test_histogram.py:17]: I am not sure I understand this one. All the functions are named "test_histogram_xxx" where xxx is more information on what the test is all about.

File: tests/test_histogram.py (hunk 3 above)
Comment (khalidcawl) [reply tests/test_histogram.py:50]: Thanks for catching that! Fixed.

File: tests/test_histogram.py (hunk 2 above)
Comment (Anthea98) [reply tests/test_histogram.py:18]: okok, I think I didn't understand Tiff

**Review by Anthea98 — APPROVED** [review]
Nice work!
